### PR TITLE
Add config resolver

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

### CODE REVIEW

**Comment by kalaiyarasiganeshalingam** [comment]
Closed this due to  It being open for more than 14 days. I will open it when we are ready to review.

File: modules/core/src/main/java/org/apache/synapse/util/synapse/expression/ast/HeadersAndPropertiesAccessNode.java [hunk 1]
```diff
@@ -60,6 +61,8 @@ public ExpressionResult evaluate(EvaluationContext context) {
             Object value;
             if (Type.HEADER.equals(type)) {
                 value = context.getHeader(name);
+            } else if (Type.CONFIG.equals(type)) {
+                value =  PropertyHolder.getInstance().getPropertyValue(name);
```
Comment (GDLMadushanka) [inline modules/core/src/main/java/org/apache/synapse/util/synapse/expression/ast/HeadersAndPropertiesAccessNode.java:65]: handle exception

File: modules/commons/src/main/java/org/apache/synapse/commons/util/FilePropertyLoader.java [hunk 2]
```diff
@@ -41,6 +42,7 @@ public class FilePropertyLoader {
     private static final String CONF_LOCATION = "conf.location";
     private static final String FILE_PROPERTY_PATH = "properties.file.path";
     private static final String DEFAULT_PROPERTY_FILE = "file.properties";
+    private static final String CONFIG_PROPERTY_FILE = "config.properties";
```
Comment (GDLMadushanka) [inline modules/commons/src/main/java/org/apache/synapse/commons/util/FilePropertyLoader.java:45]: un-used

File: modules/commons/src/main/java/org/apache/synapse/commons/resolvers/ResolverFactory.java [hunk 3]
```diff
@@ -147,4 +116,19 @@ private void registerExterns() {
...
+
+    private Resolver getResolver(String className, Matcher matcher) {
+        Class<? extends Resolver> resolverClass = resolverMap.get(className);
+        if (resolverClass != null) {
+            try {
+                Resolver resolverObject = resolverClass.newInstance();
+                resolverObject.setVariable(matcher.group(3));
+                return resolverObject;
+            } catch (IllegalAccessException | InstantiationException e) {
+                throw new ResolverException("Resolver could not be found");
```
Comment (GDLMadushanka) [inline modules/commons/src/main/java/org/apache/synapse/commons/resolvers/ResolverFactory.java:128]: throw new ResolverException("Resolver could not be initialized", e);

File: modules/core/src/main/java/org/apache/synapse/util/synapse/expression/ast/HeadersAndPropertiesAccessNode.java [hunk 4]
```diff
@@ -60,6 +61,12 @@ public ExpressionResult evaluate(EvaluationContext context) {
             Object value;
             if (Type.HEADER.equals(type)) {
                 value = context.getHeader(name);
+            } else if (Type.CONFIG.equals(type)) {
+                try {
+                    value = PropertyHolder.getInstance().getPropertyValue(name);
+                } catch (Exception e) {
```
Comment (GDLMadushanka) [inline modules/core/src/main/java/org/apache/synapse/util/synapse/expression/ast/HeadersAndPropertiesAccessNode.java:67]: Don't catch general exceptions. You can catch ResolverException here

**Review by GDLMadushanka — APPROVED** [review]

**Pull request merged**